sos/plugins/cnv.py

```diff
@@ -0,0 +1,234 @@
+            else:
+                out = line.split()[7]
+        return out
+
+    def setup(self):
+        oc_admin_cfg = os.path.join(self.master_base_dir, "admin.kubeconfig")
+        oc_cmd = "%s --config=%s" % ("oc", oc_admin_cfg)
+
+        all_dep = [
+            "%s get deployment -l kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_svc = [
+            "%s get service -l kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get service -l cdi.kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_pods = [
+            "%s get pods -l kubevirt.io=virt-controller "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l kubevirt.io=virt-api "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l cdi.kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l app=containerized-data-importer "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_stor = [
+            "%s get pvc --all-namespaces -o wide" % oc_cmd,
+        ]
+
+        subcmds = [
+            "nodes",
+            "clusterroles",
+            "clusterrolebinding",
+            "serviceaccount --all-namespaces",
+            "storageclass --all-namespaces",
+            "pv --all-namespaces",
+            "dv --all-namespaces",
+        ]
+        subcmds_dep = [
+            "describe deployment",
+        ]
+        subcmds_svc = [
+            "describe svc",
+        ]
+        subcmds_pods = [
+            "describe pod",
+            "logs",
+        ]
+
+        self.add_cmd_output(["%s version" % oc_cmd])
+        self.add_cmd_output(["virtctl --kubeconfig %s version" % oc_admin_cfg])
+        for s in subcmds:
+            self.add_cmd_output(["%s " % oc_cmd + "get " + s])
+        for dep in all_dep:
+            for d in self.get_resc_ns(self.get_cmd_output_now(dep)):
+                for s in subcmds_dep:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + d])
+        for svc in all_svc:
+            for sv in self.get_resc_ns(self.get_cmd_output_now(svc)):
+                for s in subcmds_svc:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + sv])
+        for pod in all_pods:
+            for p in self.get_resc_ns(self.get_cmd_output_now(pod)):
+                for s in subcmds_pods:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + p])
+        for stor in all_stor:
+            for pvc in self.get_resc_ns(self.get_cmd_output_now(stor)):
+                self.add_cmd_output(["%s get pvc -o yaml " % oc_cmd + pvc])
+        self.add_cmd_output("%s get vm --all-namespaces" % oc_cmd)
+        vmis = self.get_cmd_output_now("%s get vmi --all-namespaces" % oc_cmd)
+
+        if self.get_option("vm"):
+            options = self.get_option("vm")
+            if isinstance(options, bool):
+                return
```

Comment:
So just to be clear,  unless a user explicitly runs `sosreport -k cnv.vm=foobar` we don't want to do _any_ vm/vmi related collection?